crates/ruff_python_formatter/src/comments/visitor.rs

```diff
@@ -512,13 +529,29 @@ impl<'a> CommentPlacement<'a> {
     }
 }
 
+pub(super) trait CommentsBuilderTrait<'a> {
```

Comment:
It's uncommon to include the type-kind in the name, especially because only the `CommentsBuilder` is building `Comments`. 

A rusty name could be `AddComment`, or we use `Add<DecoratedComment>` and make `locator` a field on `CommentsBuilder`. 

Or another idea is to accept any type that implements `Extend`. The downside is that we'll have to pass `[comment]` because `extend_one` isn't stable. But I think that resembles the idea the closest. You get a comment and can append it to some data structure. Which also removes the need for `CommentsCollector` because you can simply pass a `&mut Vec`.

I think I'm then leaning to making `collect_comments` public and make it accept any `&dyn Extend<DecoratedComment<'a>>` and to make  `CommentsBuilder` pub(super)`.

Reply:
I went with `PushComment::push_comment`, mirroring `Vec`. A possible improvement could be implementing `PushComment` directly on `CommentsMap<'a>` and `Vec<DecoratedComment<'a>>`. I think we do need a dedicated trait though.

Reply:
I don't mind implementing it directly on `Vec<DecoratedComment>` but I think I would keep the builder around since we don't want adding the `Locator` to `CommentsMap`.